### PR TITLE
NESO v0.1.0

### PR DESCRIPTION
**Pull request by will-saunders-ukaea**
Staging branch for v0.1.0.

### CODE REVIEW

**Comment by giordano** [comment]
I was going to open a PR to add newer versions of neso-particles to include the fix for https://github.com/ExCALIBUR-NEPTUNE/NESO-Particles/issues/37 before I noticed there was already this PR :slightly_smiling_face:

**Comment by will-saunders-ukaea** [comment]
We should get NESO PRs: https://github.com/ExCALIBUR-NEPTUNE/NESO/pull/225 (for the namespace std fixes) and https://github.com/ExCALIBUR-NEPTUNE/NESO/pull/191 (for the SYCL GPU fixes) merged. Then we are in a position to make a v0.1.0 NESO.

**Comment by oparry-ukaea** [comment]
Spack env doesn't concretize for me because of this change I made in NESO-Spack...
https://github.com/ExCALIBUR-NEPTUNE/NESO-Spack/pull/36/commits/fe6c7e533909adebeb04abbff3e83cd7388c6a7c
@cmacmackin - Is there any reason not to just change this to
`depends_on("nektar+compflow_solver", type="link")`
?

**Comment by oparry-ukaea** [comment]
We're still specifying nektar@5.2.0-2022-09-03 in spack.yaml, because NESO@0.1.0 won't include the updates for nektar v5.5

**Comment by cmacmackin** [comment]
Can't you just specify `nektar@master` in the spack.yaml file?

**Comment by oparry-ukaea** [comment]
That would mean the nektar patch for 5.2 wouldn't be applied

**Comment by oparry-ukaea** [comment]
Don't think so, unfortunately.

[Sounds like](https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison) spack interprets "5.2.0-2022-09-03" as a pre-release of 5.2.0? I missed that when we were deciding on a label...

If I change it to "5.3.0-2022-09-03" then I think we can just do

`depends_on("nektar@5.3.0-2022-09-03:+compflow_solver", type="link")`
and everything will work?

**Comment by oparry-ukaea** [comment]
I've added those changes in #39; suggest we merge that to main, then main into this branch if it's all ok.

**Comment by will-saunders-ukaea** [comment]
> I've added those changes in #39; suggest we merge that to main, then main into this branch if it's all ok.

Sure. Sorry I forgot this branch existed when I made the NESO v0.1.0 PR.

File: packages/neso-particles/package.py [hunk 1]
```diff
@@ -14,8 +14,8 @@ class NesoParticles(CMakePackage):
     version("0.4.0", commit="c615974661e0f4c8d9db709d65d27cc2927bbfaf")
     version("0.3.1", commit="9c6b4626645f6aaaca478e4798f2fdee5dd2675b")
     version("0.3.0", commit="cb55184bb7196c210d8c3f4397f4a29808acd038")
-    version("0.2.0", commit="2e1b1aac6f4f9c22b31787cf300fcc2b4914cbe1")
-    version("0.1.0", commit="bae6ee91fe6558a0d7eba040dd93db34eda348a8")
+    version("0.2.0", commit="bae6ee91fe6558a0d7eba040dd93db34eda348a8")
```
Comment (oparry-ukaea) [inline packages/neso-particles/package.py:17]: @will-saunders-ukaea I got a conflict with the 0.1.0 and 0.2.0 hashes when I merged - could you quickly check and make sure they're correct?

**Review by oparry-ukaea — APPROVED** [review]
Think this is ready to merge
The latest NESO/feature/v0.1.0 builds (gcc) with this from a fresh clone and the tests pass.

File: packages/neso-particles/package.py (hunk 1 above)
Comment (will-saunders-ukaea) [reply packages/neso-particles/package.py:17]: Main looks swapped, here looks correct - see the right hand column here: https://github.com/ExCALIBUR-NEPTUNE/NESO-Particles/releases

**Pull request merged**